rcsbapi/data/schema.py

```diff
@@ -306,38 +317,37 @@ def makeFieldNode(self, parent_type: str, field_name: str) -> FieldNode:
             else:
                 index = self.schema_graph.add_child(parent_type_index, field_node, "draw")
         self.node_index_dict[field_name] = index
-        field_node.setIndex(index)
+        field_node.set_index(index)
         return field_node
 
-
-    # # maybe make this more performant
-    # def buildSchema(type_fields_dict, node_index_dict):  # iteratively
-    #     all_type_names = type_fields_dict.keys()
-    #     for type_name in all_type_names:
-    #         # print(type_name)
-    #         field_list = type_fields_dict[type_name].keys()
-    #         # if the node is already in the graph, retrieve the node. Else, make the node
-    #         if type_name not in node_index_dict.keys():
-    #             makeTypeNode(type_name)
-    #         # type_node_index = type_node.index
-    #         for field_name in field_list:
-    #             return_type = type_fields_dict[type_name][field_name]['kind'].upper()
-    #             # makeFieldNode(type_name, field_name)
-    #             if return_type == 'SCALAR':
-    #                 pass
-    #                 # field_type_name = type_fields_dict[type_name][field_name]['name'] #get the name of scalar type?
-    #             if return_type == 'OBJECT':
-    #                 makeFieldNode(type_name, field_name)  # TODO: comment out and uncomment above later
-    #                 field_type_name = type_fields_dict[type_name][field_name]['name']
-    #                 if field_type_name not in node_index_dict.keys():
-    #                     makeTypeNode(field_type_name)
-    #                 field_type_index = node_index_dict[field_type_name]
-    #                 field_index = node_index_dict[field_name]
-    #                 schema_graph.add_edge(field_index, field_type_index, None)
-    #             if return_type == 'LIST' or return_type == 'NON_NULL':
-    #                 makeFieldNode(type_name, field_name)   # TODO: comment out and uncomment above later
-    #                 field_type_name = type_fields_dict[type_name][field_name]['name']
-
+    def verify_unique_field(self, return_data_name):
+        split_data_name = return_data_name.split('.')
+        if len(split_data_name) == 1:
+            field_name = split_data_name[0]
+        else:
+            field_name = split_data_name[1]      
+        name_count = self.field_names_list.count(field_name)
+        if name_count == 1:
+            return True
+        if name_count > 1:
+            return False
+        if name_count == 0:  # TODO: if count == 0, what to do? Do I want to throw an error here? it should be caught at other points 
+            return None
+
+    def check_input_format(self, return_data_input):
+        if isinstance(return_data_input, str):
+            pass
```

Comment:
return_data will be a list so I think we can only check for list instance and throw an error when it's not a list